TwoDo/app/src/main/java/com/example/twodo/TwoDoActivity.kt

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License.
+ */
+
+package com.example.twodo
+
+import android.content.Context
+import android.os.Bundle
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+import androidx.appcompat.app.AppCompatActivity
+import androidx.fragment.app.FragmentContainerView
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.slidingpanelayout.widget.SlidingPaneLayout
+
+enum class TwoDoFragments {
+    SPLASH, EDIT
+}
+
+class TwoDoActivity : AppCompatActivity(), TaskAdapter.TaskEvents {
+
+    lateinit var slidingPane: SlidingPaneLayout
+
+    private lateinit var taskContainer: FragmentContainerView
+
+    private lateinit var taskViewModel: TaskViewModel
+    private lateinit var taskAdapter: TaskAdapter
+
+    // current fragment in FragmentContainerView
+    private var currentFragment = TwoDoFragments.SPLASH
+
+    companion object {
+        const val FRAGMENT_NAME = "FRAGMENT_NAME"
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_two_do)
+
+        taskAdapter = TaskAdapter(this)
+
+        // set up recyclerview
+        val rclList = findViewById<RecyclerView>(R.id.recycler_view)
+        rclList.adapter = taskAdapter
+        rclList.layoutManager = LinearLayoutManager(this)
+        rclList.addItemDecoration(DividerItemDecoration(rclList.context, DividerItemDecoration.VERTICAL))
+
+        taskContainer = findViewById(R.id.task_container)
+        showSplashFragment()
+
+        // set up viewmodel
+        taskViewModel = ViewModelProvider(this).get(TaskViewModel::class.java)
+        // when the task list changes, update the recyclerview
+        taskViewModel.getTaskList().observe(
+            this,
+            Observer { task ->
+                task.let { updateTaskList(task) }
+            }
+        )
+
+        slidingPane = findViewById(R.id.sliding_pane_layout)
+        val fab: View = findViewById(R.id.fab)
+        fab.setOnClickListener { view ->
+            taskViewModel.editing = false
+            showNewEditFragment()
+        }
```

Comment:
Here, instead of using comments to explain what each specific portion of code does, and empty lines to separate that different functionality, you could extract each functionality into small functions so then later from here we can read it easily:
```
    override fun onCreate(savedInstanceState: Bundle?) {
         ...
              setupThis()
              thenDothat()
              finishWithThis()
         }
```